### PR TITLE
OU-138: Move QueryBrowser component to dynamic SDK

### PR DESCRIPTION
**Pull request by kyoto**
Move the `QueryBrowser` component and `queryBrowserTheme` under `console-shared/` and expose from the SDK.

Move `QueryBrowserProps` and `FormatSeriesTitle` types to SDK.

Clean up exports / imports by remove 3 exports from the QueryBrowser file:
- Move `QueryObj` type to `dev-console` package since it is only used in that package.
- Remove `colors` and the `Error` component because they are very simple and don't really add value.

Fix eslint errors:
- Changed 2 `key` props to fix eslint `react/no-array-index-key` errors
- Rename `QueryBrowser_` to `QueryBrowserWrapped` and `Tooltip_` to `TooltipWrapped` to fix eslint `no-underscore-dangle` errors
- Several other small code changes were automatically made by eslint

### CODE REVIEW

**Comment by kyoto** [comment]
This impacts the user facing API, so requesting approvals.

/cc @RickJWagner
For PX approval

/cc @opayne1
For docs approval

File: frontend/packages/console-dynamic-plugin-sdk/docs/api.md [hunk 1]
```diff
@@ -2184,6 +2185,63 @@ Creates full page ErrorBoundaryFallbackPage component to display the "Oh no! Som
...
+| `defaultSamples` | (optional) The default number of data samples plotted for each data series. If there are many data series, QueryBrowser may automatically pick a lower number of data samples than specified here. |
+| `defaultTimespan` | (optional) The default timespan for the graph in milliseconds - defaults to 1,800,000 (30 minutes). |
+| `disabledSeries` | (optional) Disable (don't display) data series with these exact label / value pairs. |
+| `disableZoom` | (optional) Flag to disable the graph zoom controls. |
+| `filterLabels` | (optional) Optionally filter the returned data series to only those that match these label / value pairs. |
+| `fixedEndTime` | (optional) Set the end time for the displayed time range rather than showing data up to the current time. |
+| `formatSeriesTitle` | (optional) Function that returns a string to use as the title for a single data series. |
+| `GraphLink` | (optional) Component for rendering a link to another page (e.g. for getting more information about this query). |
+| `hideControls` | (optional) Flag to hide the graph controls for changing the graph timespan, etc. |
```
Comment (opayne1) [inline frontend/packages/console-dynamic-plugin-sdk/docs/api.md:2230]:
```suggestion
| `hideControls` | (optional) Flag to hide the graph controls for changing the graph timespan, and so on.
```

**Review by opayne1 — COMMENTED** [review]
Just a few small nits to adhere to the style guides.

**Comment by jhadvig** [comment]
@kyoto could you please rebase this one. Thanks ! :)

**Comment by kyoto** [comment]
@jhadvig Rebased

File: frontend/packages/console-dynamic-plugin-sdk/docs/api.md [hunk 2]
```diff
@@ -2184,6 +2185,63 @@ Creates full page ErrorBoundaryFallbackPage component to display the "Oh no! Som
...
+
+
+
+
+### Parameters
+
+| Parameter Name | Description |
+| -------------- | ----------- |
+| `customDataSource` | (optional) Base URL of an API endpoint that handles PromQL queries. If provided, this is used instead of the default API for fetching data. |
+| `defaultSamples` | (optional) The default number of data samples plotted for each data series. If there are many data series, QueryBrowser may automatically pick a lower number of data samples than specified here. |
```
[the remark this answers is not in the document]
Comment (kyoto) [reply frontend/packages/console-dynamic-plugin-sdk/docs/api.md]: 👍 I think "might" works well here

File: frontend/packages/console-dynamic-plugin-sdk/docs/api.md [hunk 3]
```diff
@@ -2184,6 +2185,63 @@ Creates full page ErrorBoundaryFallbackPage component to display the "Oh no! Som
...
+| `defaultTimespan` | (optional) The default timespan for the graph in milliseconds - defaults to 1,800,000 (30 minutes). |
+| `disabledSeries` | (optional) Disable (don't display) data series with these exact label / value pairs. |
+| `disableZoom` | (optional) Flag to disable the graph zoom controls. |
+| `filterLabels` | (optional) Optionally filter the returned data series to only those that match these label / value pairs. |
+| `fixedEndTime` | (optional) Set the end time for the displayed time range rather than showing data up to the current time. |
+| `formatSeriesTitle` | (optional) Function that returns a string to use as the title for a single data series. |
+| `GraphLink` | (optional) Component for rendering a link to another page (e.g. for getting more information about this query). |
```
[the remark this answers is not in the document]
Comment (kyoto) [reply frontend/packages/console-dynamic-plugin-sdk/docs/api.md]: 👍

File: frontend/packages/console-dynamic-plugin-sdk/docs/api.md (hunk 1 above)
Comment (kyoto) [reply frontend/packages/console-dynamic-plugin-sdk/docs/api.md]: 👍

**Comment by kyoto** [comment]
@opayne1 Addressed your feedback. Could you please check again?

**Review by jhadvig — APPROVED** [review]
/lgtm

**Comment by juzhao** [comment]
bug: https://issues.redhat.com/browse/OU-190

**Review by RickJWagner — COMMENTED** [review]
Looks good.

**Comment by yapei** [comment]
did following regression testings
- Administrator perspective: Cluster utilization, project utilization, Observe -> Metrics, Observe -> Dashbboard, Workloads Metrics
- Developer perspective: Dashboard, Metrics
No issues found
/label qe-approved

**Pull request merged**